Source/WebCore/rendering/RenderLayer.cpp

```diff
@@ -5829,26 +5829,36 @@ static bool isRenderElementWithHDR(const RenderElement& renderer)
             if (cachedImage->hasPaintedHDRContent())
                 return true;
         }
-        return false;
-    }
-
-    if (CheckedPtr imageRenderer = dynamicDowncast<LegacyRenderSVGImage>(renderer)) {
+    } else if (CheckedPtr imageRenderer = dynamicDowncast<LegacyRenderSVGImage>(renderer)) {
         if (auto* cachedImage = imageRenderer->imageResource().cachedImage()) {
             if (cachedImage->hasPaintedHDRContent())
                 return true;
         }
-        return false;
-    }
-
 #if ENABLE(PIXEL_FORMAT_RGBA16F)
-    if (CheckedPtr canavsRenderer = dynamicDowncast<RenderHTMLCanvas>(renderer)) {
+    } else if (CheckedPtr canavsRenderer = dynamicDowncast<RenderHTMLCanvas>(renderer)) {
         if (auto* renderingContext = canavsRenderer->canvasElement().renderingContext()) {
             if (renderingContext->isHDR())
                 return true;
         }
-        return false;
-    }
 #endif
+    }
+
+    CheckedRef style = renderer.style();
+
+    if (style->hasBackgroundImage()) {
+        if (auto* cachedImage = style->backgroundLayers().image()->cachedImage()) {
```

Comment:
`backgroundLayers()` is a list of FillLayers; you need to traverse the list. Your test should test multiple background images.

Reply:
Fixed.